pkg/server/storage_api/nodes_test.go

```diff
@@ -83,62 +83,78 @@ func TestStatusJson(t *testing.T) {
 func TestNodeStatusResponse(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
-	srv := serverutils.StartServerOnly(t, base.TestServerArgs{
-		DefaultTestTenant: base.TestIsForStuffThatShouldWorkWithSecondaryTenantsButDoesntYet(110023),
-	})
-	defer srv.Stopper().Stop(context.Background())
-
-	if srv.TenantController().StartedDefaultTestTenant() {
-		// Enable access to the nodes endpoint for the test tenant.
-		_, err := srv.SystemLayer().SQLConn(t).Exec(
-			`ALTER TENANT [$1] GRANT CAPABILITY can_view_node_info=true`, serverutils.TestTenantID().ToUint64())
-		require.NoError(t, err)
 
-		serverutils.WaitForTenantCapabilities(t, srv, serverutils.TestTenantID(), map[tenantcapabilities.ID]string{
-			tenantcapabilities.CanViewNodeInfo: "true",
-		}, "")
+	type testCase struct {
+		name                     string
+		defaultTestTenantOptions base.DefaultTestTenantOptions
+		expectedErr              string
 	}
 
-	s := srv.ApplicationLayer()
+	testCases := []testCase{
+		{
+			name:                     "test_node_status_reponse_shared_tenant",
+			defaultTestTenantOptions: base.SharedTestTenantAlwaysEnabled,
+		},
+		{
+			name:                     "test_node_status_reponse_external_tenant",
+			defaultTestTenantOptions: base.ExternalTestTenantAlwaysEnabled,
+			expectedErr:              "status: 500", // `can_view_node_info` capability is missing
```

Comment:
Thanks for adding this test. 500 for a permissions error is sad, but that is an existing problem that I think we have an issue for somewhere.